src/java/voldemort/client/protocol/vold/VoldemortNativeClientRequestFormat.java

```diff
@@ -116,7 +116,27 @@ public void writeGetRequest(DataOutputStream outputStream,
     }
 
     public boolean isCompleteGetResponse(ByteBuffer buffer) {
-        return isCompleteResponse(buffer, VoldemortOpCode.GET_OP_CODE);
+        try {
+            DataInputStream inputStream = new DataInputStream(new ByteBufferBackedInputStream(buffer));
+            checkException(inputStream);
+            if(!skipeResults(inputStream, buffer)) {
+                return false;
+            }
+        } catch(VoldemortException e) {
+        } catch(IOException e) {
+            return false;
+        }
+        return !buffer.hasRemaining();
+    }
+
+    private boolean skipeResults(DataInputStream inputStream, ByteBuffer buffer) throws IOException {
```

Comment:
can you please add javadocs here? I get the idea that you are trying to just wait enough for the  'len' bytes to be available and just skip over.. but someone has to reconstruct this state everytime.. (on an unrelated note, @bhasudha can you throw the doc you had on VNP protocol formats out into an external wiki page? No one really knows that the wire format it unless they read code)